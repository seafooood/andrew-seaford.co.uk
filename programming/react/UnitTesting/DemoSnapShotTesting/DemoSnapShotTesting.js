const DemoSnapShotTesting = () => (
    <div>
        <div>Hello 1</div>
        <div>Hello 2</div>
    </div>
);
export default DemoSnapShotTesting;