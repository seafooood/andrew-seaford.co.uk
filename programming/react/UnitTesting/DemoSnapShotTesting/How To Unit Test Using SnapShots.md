# How To Unit Test Using SnapShots

In this article we are demostrate create a snapshot from the rendering of a component. The unit test will fail if the conponent rendering does not match the snapshot.

## Procedure

- Create the react app `npc create-react-app demo-snapshot-testing`
- Change directory `cd demo-snapshot-testing`
- Install the test renderer `npm install react-test-renderer`
- Create the file `DemoSnapShotTesting.js` in the `src` directory.
[DemoSnapShotTesting.js](DemoSnapShotTesting.js)
- Create the file `DemoSnapShotTesting.test.js` in the `src` directory. The unit test will render the component and save the output to the variable result in JSON format. The unit test will then compare the result to the snapshot.
[DemoSnapShotTesting.test.js](DemoSnapShotTesting.test.js)
- The first time the test is executed, the test will create a snapshot within the directory `__snapshots__`.
[__snapshots__\DemoSnapShotTesting.test.js.snap](__snapshots__\DemoSnapShotTesting.test.js.snap)
- Run the tests `npm test`
