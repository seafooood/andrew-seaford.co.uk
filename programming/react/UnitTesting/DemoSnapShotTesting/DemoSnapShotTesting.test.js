import DemoSnapShotTesting from './DemoSnapShotTesting';
import TestRenderer from 'react-test-renderer'; 

test("Should check component output against snapshot", () => {
    // Arrnag

    // Act
    const result = TestRenderer.create(<DemoSnapShotTesting />).toJSON();

    // Assert
    expect(result).toMatchSnapshot();
});