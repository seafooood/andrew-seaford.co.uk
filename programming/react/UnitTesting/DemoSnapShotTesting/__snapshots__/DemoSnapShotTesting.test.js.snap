// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should check component output against snapshot 1`] = `
<div>
  <div>
    Hello 1
  </div>
  <div>
    Hello 2
  </div>
</div>
`;
